Fix a fair amount of Elvis warnings

cjsthompson opened the pull request:
Mostly "implicit declaration" warnings. Add appropriate headers to fix this. Also reformat some function declarations to ANSI C and some comments to fit in 80 columns.

One "implicit declaration" warning that is still left is for "tmpnam" whose declaration is missing in ELKS' headers. Not sure whether this has to go in libc or gcc headers. In Linux it's declared in stdio.h according to the man page.

Elvis still compiles and runs and edits in ELKS.

dptpirate committed:
Fix a fair amount of Elvis warnings

Also reformated some function declarations to ANSI C and some comments
to fit in 80 columns.

ghaerr commented:
Nice work Christoph! And elvis had a ton of warnings. There may be a day yet where ELKS application compilation warnings are no longer, and this is a big step forward!

> One "implicit declaration" warning that is still left is for "tmpnam" whose declaration is missing in ELKS' headers. Not sure whether this has to go in libc or gcc headers. In Linux it's declared in stdio.h according to the man page.

Since ELKS libc/misc has tmpnam.c, and that's being used, go ahead and add the tmpnam prototype to libc/include/stdio.h, and that'll be one less to worry about. I'll commit this after you push that additional fix.

A quick search shows tmpnam is used in elvis, sys_utils/passwd.c, and busyelks/cmd/passwd.c, and they include <stdio.h>.

dptpirate committed:
Declare tmpnam() in stdio.h and change declaration
of tmpnam() to ANSI C.

	modified:   libc/include/stdio.h
	modified:   libc/misc/tmpnam.c

cjsthompson commented:
Here it is. I've got more easy warning fixes. But I'd thought I'd group them by module instead of doing a massive pull request that is harder to review.

ghaerr commented:
Ok, I'll commit this now. In the future, we're not using the older "extern ..." in libc header files, instead just the straight prototype. You can fix extern char *tmpnam() in a future commit.

cjsthompson commented:
I will. Thanks.
